Rails 4.2 compatibility

sometoby opened the pull request:
This PR updates the plugin to be compatible with Rails 4. It also fixes a minor problem where bulk updating issues wouldn't display the number of updated issues correctly.

Commit:
Update query methods from Rails 3 to Rails 4

Commit:
Permit parameters for creating and editing Deliverables

Commit:
Fix number of updated issues not being shown correctly

Commit:
Remove trailing whitespace and fix a typo

Commit:
Fix links and routes for Rails 4

Commit:
Fix a wrong query method

Commit:
Use find instead of find_by_id

This rewrite also stops the hook from making a pointless db query when
detail.value or detail.old_value are already the Deliverable's subject.

Commit:
Fix some broken migrations

Commit:
Remove unloadable calls to fix autoloading

http://www.redmine.org/issues/20513

Commit:
Remove empty files

felixbuenemann commented:
I copied over the `redmine-3.1` branch, but have not yet merged it, because I think we still have apps currently referencing `master` and I should create a pre-3.1 branch for them first.

Commit:
Guard against context not having a project

Commit:
"Fix" deliverables in bulk copies

This seems like a hack, and it probably is. I don't know how else to get
at the target project from a hook.
